typing/typecore.ml

```diff
@@ -2000,7 +2000,7 @@ let create_package_type loc env (p, l) =
    let open Ast_helper in
    List.fold_left
      (fun sexp (name, loc) ->
-       Exp.letmodule ~loc:sexp.pexp_loc
+        Exp.letmodule ~loc:{ sexp.pexp_loc with loc_ghost = true }
```

Comment:
Why is the location ghost? Is it because it does not correspond to an AST node proper, but only part of it (the module name)?
I guess it makes some sense, but it contradicts my existing mental model of ghost locations, where for me they would be used for pieces of code that were elaborated and do not correspond to user syntax.
Does that make sense?

Reply:
The location is ghost precisely because there is elaboration, and if I'm not mistaken `sexp.pexp_loc` points to the rhs of the arrow, e.g. `| (module M : S) -> sexp`.

Reply:
Can you confirm this fix is now independent from the PR main task?

Reply:
It is now independent from the rest of the PR indeed.

Reply:
Thanks for the confirmation.  As requested, I let you take care of merging.